adapter/internal/notifier/deployment_notifier.go

```diff
@@ -82,3 +83,51 @@ func SendRevisionUpdate(deployedRevisionList []*DeployedAPIRevision) {
 		}
 	}
 }
+
+// SendRevisionUndeploy - send the undeployed revision information to control plane
+func SendRevisionUndeploy(apiUUID string, revisionUUID string, environment string) {
+	conf, _ := config.ReadConfigs()
+	cpConfigs := conf.ControlPlane
+
+	revisionEP := cpConfigs.ServiceURL
+	if strings.HasSuffix(revisionEP, "/") {
+		revisionEP += unDeployedRevisionEP
+	} else {
+		revisionEP += "/" + unDeployedRevisionEP
+	}
+
+	if apiUUID == "" || revisionUUID == "" || environment == "" {
+		return
+	}
+
+	removedRevision := UnDeployedAPIRevision{
+		APIUUID:      apiUUID,
+		RevisionUUID: revisionUUID,
+		Environment:  environment,
+	}
+
+	jsonValue, _ := json.Marshal(removedRevision)
+	basicAuth := "Basic " + auth.GetBasicAuth(cpConfigs.Username, cpConfigs.Password)
+	retries := 0
+	for retries < 3 {
+		retries++
+		req, _ := http.NewRequest("POST", revisionEP, bytes.NewBuffer(jsonValue))
+		req.Header.Set("Authorization", basicAuth)
+		req.Header.Set("Content-Type", "application/json")
+		resp, err := tlsutils.InvokeControlPlane(req, cpConfigs.SkipSSLVerification)
+
+		success := true
+		if err != nil {
+			logger.LoggerNotifier.Warnf("Error response from %v for attempt %v : %v", revisionEP, retries, err.Error())
+			success = false
+		}
+		if resp != nil && resp.StatusCode != http.StatusOK {
+			logger.LoggerNotifier.Warnf("Error response status code %v from %v for attempt %v", resp.StatusCode, revisionEP, retries)
+			success = false
+		}
```

Comment:
Would not it be better to have at least one second delay before retrying. Or may be 5 secs.
@AmaliMatharaarachchi  wdyt?

Reply:
Added 2 seconds delay.